### ISSUE TITLE
Aggressive fingerprinting mode removal

### ISSUE THREAD

**Issue by arthuredelstein** [brave/brave-browser#31229]
We are considering removing aggressive fingerprinting mode, to reduce unneeded complexity in browser code and in the UI. To do this, we would need to work out which protections from aggressive mode to drop, and which ones to fold into default fingerprinting mode. Current protections in aggressive mode include:

 * Dark mode detection: https://github.com/brave/brave-browser/issues/15265
 * WebGL: https://github.com/brave/brave-browser/issues/9188, https://github.com/brave/brave-browser/issues/15882 
 * navigator.userAgent farbling: https://github.com/brave/brave-core/pull/6055/

**Comment by IronMaltese** [comment]
It would be nice to have Dark Mode protection as an option in the shields settings

**Comment by IronMaltese** [comment]
What's the plan for the other settings? Is there no possibility that we can keep them in default mode, but disable them for sites that break?

**Comment by IronMaltese** [comment]
I can't wait for the blog post to see how these are resolved, I'm quite curious

**Comment by stephendonner** [comment]
Hi @arthuredelstein ! Mind adding a testplan here, when you get a chance? Thanks! Adding QA/Blocked and QA/Test-Plan-Required just until we've got that, and then we'll be fully unblocked 👍

**Comment by stephendonner** [comment]
## Verification `IN-PROGRESS` using

```
Brave | 1.62.59 Chromium: 119.0.6045.163 (Official Build) nightly (x86_64)
-- | --
Revision | e1b8145b091a14990d50127dc7ac0c407f8a13a6
OS | macOS Version 14.2 (Build 23C5047e)
```

**Comment by MadhaviSeelam** [comment]
Verification `PASSED` using 
```
Brave | 1.62.105 Chromium: 120.0.6099.71 (Official Build) beta (64-bit)
-- | --
Revision | f72c783bcd52110d026061575b4bef28ccb547f7
OS | Windows 11 Version 22H2 (Build 22621.2715)
```

1. Install `1.62.105`
2. launched Brave 
3. opened `brave://flags` in a new tab
4. disabled `#brave-show-strict-fingerprinting-mode` flag
5. opened brave://settings/shields
6. navigated to brave.com site and clicked on Shields 

#### Confirmed `Block fingerprinting` toggle is enabled as a default
#### Confirmed `Block fingerprinting` toggle is enabled as a default in Per site Shields setting

step 4 | step 5 | step 6
----- | --- | -----
<img width="962" alt="image" src="https://github.com/brave/brave-browser/assets/98358127/84f15995-600f-44a1-8531-a0a3de029e81">|<img width="955" alt="image" src="https://github.com/brave/brave-browser/assets/98358127/fa090ddc-646d-4e57-8ada-f169c73f7352">|<img width="956" alt="image" src="https://github.com/brave/brave-browser/assets/98358127/958a7f11-8dcd-4616-ba84-5b98d1510801">



</details>

### https://dev-pages.bravesoftware.com/fingerprinting/strict-mode.html - `PASSED`
<details>

### Steps:
1. installed both `1.61.101` and `1.62.105`
2. launched Brave (`release`)
3. ran the tests on https://dev-pages.bravesoftware.com/fingerprinting/strict-mode.html  and https://dev-pages.brave.software/fingerprinting/strict-mode.html by clicking `Generate Values`
4. noted the results
5. compared `1.61.101` to `1.62.105`

#### Confirmed identical results for the farbled values shown (values are the same for each version, not between versions, to be clear)

#### `1.62.105`

example | example
---------|---------
<img width="955" alt="image" src="https://github.com/brave/brave-browser/assets/98358127/69e05c7b-7a41-4d97-987f-46e27391455f">|<img width="958" alt="image" src="https://github.com/brave/brave-browser/assets/98358127/c19d5462-ee6e-4df8-8858-3ff98e9a1339">

#### `1.61.101`
example | example
---------|---------
<img width="955" alt="image" src="https://github.com/brave/brave-browser/assets/98358127/2a967084-2089-4402-ab8e-c32eb9467bb5">|<img width="958" alt="image" src="https://github.com/brave/brave-browser/assets/98358127/ef42c20c-845b-4986-b03d-6cc58a97eca2">
</details>

## Upgrades from `1.61.x` - `PASSED`
<details>


### Shared Steps:

1. installed `1.61.101`
7. launched Brave (`release`)
8. opened `brave://settings/shields`
9. ensured `Block fingerprinting` value was set appropriately for each `Standard`, `Disabled`, and `Strict` case
10. shut down Brave
11. installed `1.62.105`
12. renamed `Brave-Browser` user profile --> `Brave-Browser-Beta`
13. launched Brave (`Beta`)
14. set `brave://flags/#brave-show-strict-fingerprinting-mode` to `Disabled`
15. click `Relaunch`
16. opened `brave://settings/shields`

`brave://flags`(default) | `brave://flags` (disabled) | `brave://version` | brave://settings/shields
---------------|-----------------|--------|-------
<img width="955" alt="image" src="https://github.com/brave/brave-browser/assets/98358127/2071ddde-b1e4-4cf5-9314-968fc2f4a359">|<img width="954" alt="image" src="https://github.com/brave/brave-browser/assets/98358127/7d2c8a86-975c-4e72-928b-0fd2c9dca639">|<img width="950" alt="image" src="https://github.com/brave/brave-browser/assets/98358127/1ea625dd-00db-45a2-aa6c-de64109447a4">|<img width="955" alt="image" src="https://github.com/brave/brave-browser/assets/98358127/90687eba-a96b-41ca-b8fc-edc9c0122095">

`Block fingerprinting` - `Strict, may break sites` - `PASSED`

`1.61.101` | `1.62.105 (a)`| `1.62.105 (b)`| `1.62.105 (c)`|`1.62.105 (d)`
---------|---------|---------|-------|------
<img width="939" alt="image" src="https://github.com/brave/brave-browser/assets/98358127/db72b1ec-b84d-4b35-86cb-e71816d5d0a2">|<img width="954" alt="image" src="https://github.com/brave/brave-browser/assets/98358127/2fa2e531-5448-4213-b03c-d9629bece76a">|<img width="948" alt="image" src="https://github.com/brave/brave-browser/assets/98358127/7d35e95a-4daf-4ea9-94c9-c831f3e6a4f8">|<img width="948" alt="image" src="https://github.com/brave/brave-browser/assets/98358127/61b496ce-3bfd-46e3-ab41-f44d747529c3">|<img width="954" alt="image" src="https://github.com/brave/brave-browser/assets/98358127/5dc97d15-71ef-4670-835c-4a4e6dbcf183">

`Block fingerprinting` - `Standard` - `PASSED`

`1.61.101` | `1.62.105 (a)`| `1.62.105 (b)`| `1.62.105 (c)`|`1.62.105 (d)`
---------|---------|---------|-------|------
<img width="955" alt="image" src="https://github.com/brave/brave-browser/assets/98358127/7f46a7bb-9f69-4913-b642-9737bc8a0617">|<img width="949" alt="image" src="https://github.com/brave/brave-browser/assets/98358127/42da9bbb-cb02-4347-8d7b-f84148411e5b">|<img width="953" alt="image" src="https://github.com/brave/brave-browser/assets/98358127/0c8ce554-0f36-4064-b74c-6031a77397b8">|<img width="957" alt="image" src="https://github.com/brave/brave-browser/assets/98358127/c0c7a21d-7713-44e2-9c58-57a6dddbab38">|<img width="952" alt="image" src="https://github.com/brave/brave-browser/assets/98358127/37198ee3-a342-464f-a4e2-67408fc79519">


`Block fingerprinting` - `Disabled` - `PASSED`

`1.61.101` | `1.62.105 (a)`| `1.62.105 (b)`| `1.62.105 (c)`|`1.62.105 (d)`
---------|---------|---------|-------|------
<img width="956" alt="image" src="https://github.com/brave/brave-browser/assets/98358127/b5191cb1-5a8c-4e92-baef-a70fe3828a3b">|<img width="953" alt="image" src="https://github.com/brave/brave-browser/assets/98358127/95153d72-1358-4481-bb1f-4ccd71809891">|<img width="956" alt="image" src="https://github.com/brave/brave-browser/assets/98358127/e86cf2f0-3b3f-4a3b-b29e-0afc8ab7354c">|<img width="950" alt="image" src="https://github.com/brave/brave-browser/assets/98358127/0ee3eef1-14a2-484d-8e18-29b1f6d7b8fd">|<img width="959" alt="image" src="https://github.com/brave/brave-browser/assets/98358127/86be3736-c5d5-43cb-b084-5265d6dcb8a7">





</details>

**Comment by btlechowski** [comment]
Verified with

Brave | 1.62.122 Chromium: 120.0.6099.144 (Official Build) beta (64-bit)
-- | --
Revision | cfddebe77d394064c472fda64afcd9fbed34ceb4
OS | Linux



#### New Defaults - `PASSED`
<details>

`brave://settings/Shields` | `Shields'` `Advanced Controls`
--------------------------|-------------------------------
![image](https://github.com/brave/brave-browser/assets/34715963/c5cdc3b4-e800-46f6-95c6-862575a80362)|![image](https://github.com/brave/brave-browser/assets/34715963/4ea2deec-6729-44fa-b458-e26e1660911d)




</details>

#### https://dev-pages.bravesoftware.com/fingerprinting/strict-mode.html - `PASSED`
<details>

### Steps:
1. installed both `1.61.x` and `1.62.x`
2. launched Brave (`release`)
3. ran the tests on https://dev-pages.bravesoftware.com/fingerprinting/strict-mode.html  and https://dev-pages.brave.software/fingerprinting/strict-mode.html by clicking `Generate Values`
4. noted the results
5. compared `1.61.x` to `1.62.x`

### Confirmed identical results for the farbled values shown (values are the same for each version, not between versions, to be clear)

### `1.61.x`

example | example
---------|---------
![image](https://github.com/brave/brave-browser/assets/34715963/00a765a3-d697-4c09-93b8-4e981c033bf9)|![image](https://github.com/brave/brave-browser/assets/34715963/ac6cdab6-e773-44d1-ae89-0a4cb0c5bb78)


### `1.62.x`

example | example
---------|---------
![image](https://github.com/brave/brave-browser/assets/34715963/e481f60d-6efb-4132-9709-555f27e0bda0)|![image](https://github.com/brave/brave-browser/assets/34715963/47d939ae-f047-4488-8d89-430d8695d048)


</details>

#### Upgrades from `1.61.x` - `PASSED`
<details>


### Shared Steps:

1. installed `1.61.x`
7. launched Brave (`release`)
8. opened `brave://settings/shields`
9. ensured `Block fingerprinting` value was set appropriately for each `Standard`, `Disabled`, and `Strict` case
10. shut down Brave
11. installed `1.62.x`
12. renamed `Brave-Browser` user profile --> `Brave-Browser-Beta`
13. launched Brave (`beta`)
14. set `brave://flags/#brave-show-strict-fingerprinting-mode` to `Disabled`
15. click `Relaunch`
16. opened `brave://settings/shields`






`Block fingerprinting` - `Strict, may break sites` - `PASSED`

`1.61.x` | `1.62.x`
---------|---------
![image](https://github.com/brave/brave-browser/assets/34715963/7f44e5a7-7307-47ba-8043-0f180a8844c0)|![image](https://github.com/brave/brave-browser/assets/34715963/f1867bad-3562-4717-a48d-5efbd219c44f)


`Block fingerprinting` - `Standard` - `PASSED`

`1.61.x` | `1.62.x`
---------|---------
![image](https://github.com/brave/brave-browser/assets/34715963/d1f2e487-c638-44fe-b193-aef8ff3cc8d5)|![image](https://github.com/brave/brave-browser/assets/34715963/116682fb-7c88-44b8-ba2c-8687f215d71b)



`Block fingerprinting` - `Disabled` - `PASSED`

`1.61.x` | `1.62.x`
---------|---------
![image](https://github.com/brave/brave-browser/assets/34715963/9d69d76b-4315-4ecf-bbf5-64be6a53487d)|![image](https://github.com/brave/brave-browser/assets/34715963/3401135c-c048-4264-a40b-3a8282079dd5)




</details>

**Comment by GeetaSarvadnya** [comment]
Verification **INPROGRESS** on `Vivo X70 Pro version 12` running `Bravemonoarm64.apk_1.62.121`

**Comment by Uni-verse** [comment]
In Progress on `Samsung Galaxy Tab S7` using version: 

```
Brave	1.62.147 Chromium: 120.0.6099.234 (Official Build) (32-bit) 
Revision	3b25c3743150a54485dea24f0ceb1e69d6db51bc
OS	Android 13; Build/TP1A.220624.014; 33; REL
```

**Comment by BenjaminAster** [comment]
@arthuredelstein Note that without aggressive fingerprinting protection, `UNMASKED_RENDERER_WEBGL` and `UNMASKED_VENDOR_WEBGL` from the WebGL `WEBGL_debug_renderer_info` extension will always return the true GPU info with no option to enable randomization of these values (see #10214). This is a pretty high-entropy source of fingerprintable information&mdash;shouldn't the current aggressive behavior (replacing the values with random gibberish) be folded into default fingerprinting mode? The entire canvas farbling is much less effective if websites can read the GPU type anyway.

**Comment by arthuredelstein** [comment]
@BenjaminAster @davidcollini Thank you for these comments. We definitely want to have GPU protections in standard mode.

**Comment by BenjaminAster** [comment]
@arthuredelstein Thanks for the quick response! While I'm at it, I just opened https://github.com/brave/brave-browser/issues/35646 where I have even more suggestions about anti-fingerprinting measures that I think Brave could still take.

**Comment by arthuredelstein** [comment]
Thank you @BenjaminAster !
